code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -31,37 +33,49 @@
 
 /obj/item/reagent_containers/hypospray/cmo
 	name = "premium hypospray"
-	desc = "The Zeng-Hu Pharmaceuticalsn premium hypospray is a cutting-edge, sterile, air-needle autoinjector for rapid administration of drugs to patients."
-	item_state = "cmo_hypo"
+	desc = "A high-end version of the regular hypospray, it allows for a substantially higher rate of drug administration to patients."
+	description_fluff = "The Zeng-Hu Pharmaceuticals' HypoSpray Mk-II is a cutting-edge version of the regular hypospray, with a much more expensive and streamlined injection process."
+	description_info = "This version of the hypospray has no delay before injecting a patient with reagent."
+	icon_state = "cmo_hypo"
 	volume = 30
 	time = 0
 
+/obj/item/reagent_containers/hypospray/cmo/update_icon()
+	cut_overlays()
+
+	var/rounded_vol = round(reagents.total_volume, round(reagents.maximum_volume / 6))
+	icon_state = "[initial(icon_state)]_[rounded_vol]"
+	if(reagents.total_volume)
+		filling = image('icons/obj/syringe.dmi', src, "cmo_hypo30")
+
+		filling.icon_state = "[initial(icon_state)][rounded_vol]"
+
+		filling.color = reagents.get_color()
+		add_overlay(filling)
+
```

Comment:
I'm pretty sure you can avoid this egregious case of copypaste with a simple variable or two on the hypospray.